share/completions/yadm.fish

```diff
@@ -0,0 +1,87 @@
+# Completions for yadm v2
+
+# Print an optspec for argparse to handle yadm's options
+# that are independent of any subcommand. Borrowed from git completion file.
+function __fish_yadm_global_optspecs
+    string join \n v-version h/help C= c=+ 'e-exec-path=?' H-html-path M-man-path I-info-path p/paginate \
+        P/no-pager o-no-replace-objects b-bare G-git-dir= W-work-tree= N-namespace= S-super-prefix= \
+        l-literal-pathspecs g-glob-pathspecs O-noglob-pathspecs i-icase-pathspecs \
+        Y/yadm-dir= R-yadm-repo= n-yadm-config= E-yadm-encrypt= a-yadm-archive= B-yadm-bootstrap=
+end
+
+# Borrowed from git completion file
+function __fish_yadm_needs_command
+    # Figure out if the current invocation already has a command.
+    set -l cmd (commandline -opc)
+    set -e cmd[1]
+    argparse -s (__fish_yadm_global_optspecs) -- $cmd 2>/dev/null
+    or return 0
+    # These flags function as commands, effectively.
+    set -q _flag_version; and return 1
+    set -q _flag_html_path; and return 1
+    set -q _flag_man_path; and return 1
+    set -q _flag_info_path; and return 1
+    if set -q argv[1]
+        # Also print the command, so this can be used to figure out what it is.
+        echo $argv[1]
+        return 1
+    end
+    return 0
+end
+
+# Borrowed from git completion file
+function __fish_yadm_using_command
+    set -l cmd (__fish_yadm_needs_command)
+    test -z "$cmd"
+    and return 1
+    contains -- $cmd $argv
+    and return 0
+    return 1
+end
+
+set -l yadm_work_tree (yadm gitconfig --get core.worktree)
+set -l yadm_repo (yadm introspect repo)
+
+# TODO: --yadm-repo option must override $yadm_repo calue
+complete -c yadm -w "git --work-tree $yadm_work_tree --git-dir $yadm_repo"
```

Comment:
Yeah, this won't change the work-tree at runtime. It will only be set once, when the script is loaded.

The easiest thing to do is probably to do the wrapping manually. Something like (untested):

```fish
function __fish_complete_yadm_like_git
    set -l cmdline (commandline -opc; commandline -ct)[2..-1]
    set -l yadm_work_tree (yadm gitconfig --get core.worktree)
    set -l yadm_repo (yadm introspect repo)
    complete -C"git --work-tree $yadm_work_tree --git-dir $yadm_repo $cmdline"
end

complete -c yadm -a '(__fish_complete_yadm_like_git)'
```

Reply:
Thanks for the help; it works like a charm with a few modifications. There are still a few rough edges to smooth out in other regards, and I'll finish it as soon as possible.